discord_connector.py

```diff
@@ -158,128 +162,183 @@ def _handle_test_connectivity(self, param):
+
+        attachments, embeds = None, None
+        message = self._async_loop.run_until_complete(self.fetch_message(channel_id, message_id))
+
+        # do we need to check if message is not none?
+        if message is not None:
+            if message.embeds is not None or message.attachments is not None:
+                attachments, embeds = self.create_artifacts(message)
+            message = self.parse_message(message, attachments, embeds)
+        else:
+            summary = action_result.update_summary({})
+            summary['failure: '] = "unable to fetch the message: message is None"
+            return action_result.set_status(phantom.APP_ERROR)
+
+        action_result.add_data(message)
+        summary = action_result.update_summary({})
+        summary['success: '] = "fetching message completed"
 
-        # Return success, no need to set the message, only the status
-        # BaseConnector will create a textual message based off of the summary dictionary
         return action_result.set_status(phantom.APP_SUCCESS)
 
-        # For now return Error with a message, in case of success we don't set the message, but use the summary
-        #return action_result.set_status(phantom.APP_ERROR, "Action not yet implemented")
+    async def fetch_message(self, channel_id, message_id):
+        await self._client.login(self._token)
+
+        guild = await self._client.fetch_guild(self._guild_id)
+        self.save_progress("fetched guild: {}, fetched guild id: {}".format(str(guild), str(guild.id)))
+
+        channel = await guild.fetch_channel(channel_id)
+        self.save_progress("channel: {}".format(str(channel)))
+
+        message = await channel.fetch_message(message_id)
+        self.save_progress("message: {}".format(str(message)))
+
+        await self._client.close()
+        return message
+
+    def create_artifacts(self, message):
+
+        container_id = BaseConnector.get_container_id(self)
+        attachments = []
+        embeds = []
+
+        # do we need it there or rather in consts???
+        artifact = {
+            "container_id": container_id,
+            "name": "name",
+            "cef": {
+                "URL": "",
+                "type": "",
+                "Description": ""
+            }
+        }
+
+        self.save_progress("working on embeds")
+        for embed in message.embeds:
+            self.save_progress("embed: {}".format(embed.to_dict))
+            embeds.append(self.create_embed_artifact(embed, artifact))
+
+        self.save_progress("working on attachments")
+        for attachment in message.attachments:
+            self.save_progress("attachment: {}".format(attachment.to_dict()))
+            attachments.append(self.create_attachment_artifact(attachment, artifact))
+
+        return attachments, embeds
+
+    # convert to use strategy pattern???
+
+    def create_embed_artifact(self, embed, artifact):
+        artifact["name"] = f"embed: {embed.title}"
```

Comment:
so this is basically overwriting most of the artifact dict except for container id and cef type? Maybe there's a better way to keep this information and then cast into dict if needed? 

Especially, when this is done more than once. 

maybe some [dataclass](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict)?